sdk/eventhub/azure-eventhubs/azure/eventhub/aio/_producer_client_async.py

```diff
@@ -0,0 +1,179 @@
+            async with self._producers_locks[producer_index]:
+                if self._producers[producer_index] is None:
+                    self._producers[producer_index] = self._create_producer(partition_id=partition_id)
+        async with self._producers_locks[producer_index]:
+            await self._producers[producer_index].send(event_data, partition_key=partition_key, timeout=timeout)
+
+    async def create_batch(self, max_size=None):
+        # type:(int) -> EventDataBatch
+        """
+        Create an EventDataBatch object with max size being max_size.
+        The max_size should be no greater than the max allowed message size defined by the service side.
+
+        :param max_size: The maximum size of bytes data that an EventDataBatch object can hold.
+        :type max_size: int
+        :return: an EventDataBatch instance
+        :rtype: ~azure.eventhub.EventDataBatch
+
+        Example:
+            .. literalinclude:: ../examples/test_examples_eventhub_async.py
+                :start-after: [START eventhub_producer_client_create_batch_async]
+                :end-before: [END eventhub_producer_client_create_batch_async]
+                :language: python
+                :dedent: 4
+                :caption: Create EventDataBatch object within limited size
+
+        """
+        if not self._max_message_size_on_link:
+            await self._init_locks_for_producers()
+            async with self._producers_locks[-1]:
+                if self._producers[-1] is None:
+                    self._producers[-1] = self._create_producer(partition_id=None)
+                    await self._producers[-1]._open_with_retry()  # pylint: disable=protected-access
+            async with self._client_lock:
+                self._max_message_size_on_link = \
+                    self._producers[-1]._handler.message_handler._link.peer_max_message_size or constants.MAX_MESSAGE_LENGTH_BYTES  # pylint: disable=protected-access, line-too-long
+
+        if max_size and max_size > self._max_message_size_on_link:
+            raise ValueError('Max message size: {} is too large, acceptable max batch size is: {} bytes.'
+                             .format(max_size, self._max_message_size_on_link))
+
+        return EventDataBatch(max_size=(max_size or self._max_message_size_on_link))
+
+    async def close(self):
+        # type: () -> None
+        """
+        Close down the handler. If the handler has already closed,
+        this will be a no op.
+
+        Example:
+            .. literalinclude:: ../examples/test_examples_eventhub_async.py
+                :start-after: [START eventhub_producer_client_close_async]
+                :end-before: [END eventhub_producer_client_close_async]
+                :language: python
+                :dedent: 4
+                :caption: Close down the handler.
+
+        """
+        if self._producers:
+            for p in self._producers:
+                if p:
```

Comment:
That's one of the issue to use a list filled with None and not a dict, you're not sure if you get an actual value